cloudevents/abstract/event.py

```diff
@@ -0,0 +1,97 @@
+import abc
+import typing
+from typing import TypeVar
+
+
+class CloudEvent(abc.ABC):
+    """
+    Used internally for multi-framework event type integration
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def create(
+        cls,
+        attributes: typing.Dict[str, typing.Any],
+        data: typing.Optional[typing.Any],
+    ) -> "CloudEvent":
+        """
+        Factory function.
+        We SHOULD NOT use the __init__ function as the factory because
+        we MUST NOT assume how the __init__ function looks across different python
+        frameworks
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def _attributes_read_model(self) -> typing.Dict[str, typing.Any]:
+        """
+        :return: attributes of this event
+
+        you MUST NOT mutate this dict
+        implementation MAY assume the dic will not be mutated
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def _data_read_model(self) -> typing.Optional[typing.Any]:
+        """
+        :return: data value of the  event
+        you MUST NOT mutate this value
+        implementation MAY assume the value will not be mutated
+        """
+        pass
```

Comment:
The same question here. I'd rather have it here as just `data` and hide `data` field in the implementation.

Reply:
I can't use the `data` property because in pydantic i need to declare it as a field and not a property

Reply:
Well, I'm not sure that's a valid reason to drop this approach. Maybe worth naming `data` differently in Pydantic model instead?

Reply:
but then it will be serialized funky in json form

Reply:
i think maybe we should define the cloudevent as protocol

this will make much more sense dont you think?

Reply:
Protocol would be nice, but it's only available in python 3.8+.
And we currently support 3.6+.

Reply:
> but then it will be serialized funky in json form

I guess it may be configured. E.g. by using alias for the data.

E.g. name the field "payload" and use data as alias

Reply:
good idea, i will do that

Reply:
did not work, see below, i found a better way